### ISSUE TITLE
False positive: enum became abstract

### ISSUE THREAD

**Issue by lbergelson** [lvc/japi-compliance-checker#57]
We see a false positive report of `This class became abstract` on an enum.

Previously the enum was a simple enum of the form
```
enum Thing {
  First,
  Second
}
```

It subsequently gained an abstract method:
```
enum Thing {
   First() {
     void doThing(){..}
   },
   Second(){
     void doThing(){..}
   }

  abstract void doThing(){}
}
```

We now get a false positive report the Thing is abstract despite the fact that by definition enums are concrete `final` classes.
